### PR TITLE
COD-72 Models testing

### PR DESCRIPTION
**Pull request by Xillez**
Set up test environment, tested models that actually have something valuable to test, added release buildtarget for full build with test and test coverage generation. Added test coverage generation tool to gradle.

### CODE REVIEW

File: backend/parser/build.gradle [hunk 1]
```diff
@@ -15,26 +15,49 @@ buildscript {
...
 
-
 apply plugin: 'java'
 apply plugin: 'me.champeau.gradle.antlr4'
+apply plugin: 'jacoco'
+
+jacoco {
+    //reportsDir = file("$buildDir/jacoco/reports")
+    //destinationFile = file("$buildDir/jacoco/jacocoTest.exec")
+    //classDumpDir = file("$buildDir/jacoco/classpathdumps")
```
Comment (eldarht) [inline backend/parser/build.gradle]: commented code?

File: backend/parser/build.gradle [hunk 2]
```diff
@@ -15,26 +15,49 @@ buildscript {
...
+    }
+}
+
+test.finalizedBy jacocoTestReport
 test {
     useJUnitPlatform()
+    jacoco {
+        append = false
+        destinationFile = file("$buildDir/jacoco/jacocoTest.exec")
+        //classDumpFile = file("$buildDir/jacoco/classpathdumps")
```
Comment (eldarht) [inline backend/parser/build.gradle]: commented code?

File: backend/parser/build.gradle [hunk 3]
```diff
@@ -15,26 +15,49 @@ buildscript {
     }
 }
 
-
 apply plugin: 'java'
 apply plugin: 'me.champeau.gradle.antlr4'
+apply plugin: 'jacoco'
+
+jacoco {
+    //reportsDir = file("$buildDir/jacoco/reports")
```
Comment (zohaib194) [inline backend/parser/build.gradle]: do we need these?

File: backend/parser/build.gradle (hunk 1 above)
Comment (Xillez) [reply backend/parser/build.gradle]: removed

File: backend/parser/build.gradle (hunk 2 above)
Comment (Xillez) [reply backend/parser/build.gradle]: removed

File: backend/parser/build.gradle (hunk 3 above)
Comment (Xillez) [reply backend/parser/build.gradle]: removed

**Pull request merged**